src/qililab/platform/platform.py

```diff
@@ -567,6 +590,42 @@ def __str__(self) -> str:
         """
         return str(YAML().dump(self.to_dict(), io.BytesIO()))
 
+    def execute_anneal_program(
+        self, anneal_program_dict: list[dict[str, dict[str, float]]], transpiler: Callable, averages=1
+    ):  # TODO: determine default average
+        """Given an anneal program execute it as a qprogram.
+
+        The anneal program should contain a time ordered list of circuit elements and their corresponging ising coefficients as a dictionary. Example structure:
+        [
+            {"qubit_0": {"sigma_x" : 0, "sigma_y" : 1, "sigma_z" : 2},
+            "coupler_1_0 : {...},
+            },      # time=0ns
+            {...},  # time=1ns
+        .
+        .
+        .
+        ]
+        This dictionary containing ising coefficients is transpiled to fluxes using the given transpiler. Then the correspoinding waveforms are obtained and assigned to a bus
+        from the bus to flux mapping given by the runcard.
+
+        Args:
+            anneal_program_dict (list[dict[str, dict[str, float]]]): anneal program to run
+            transpiler (Callable): ising to flux transpiler. The transpiler should take 2 values as arguments (delta, epsilon) and return 2 values (phix, phiz)
+            averages (int, optional): Amount of times to run and average the program over. Defaults to 1.
+        """
+        anneal_program = AnnealingProgram(self, anneal_program_dict)
+        anneal_program.transpile(transpiler)
+        anneal_waveforms = anneal_program.get_waveforms()
+
+        qp_anneal = QProgram()
+        with qp_anneal.average(averages):
+            for bus, waveform in anneal_waveforms.values():
+                qp_anneal.play(bus=bus.alias, waveform=waveform)
+
+        # TODO: define readout
```

Comment:
Should we include measurement as a parameter then for the execute function? what do you suggest? The question goes along where in the whole procedure we create an instance of the new measurement block and append it to this qprogram. I guess there is room for discussion with the team.

Reply:
I think measurement needs to be defined as a qprogram block / standard procedure, since it is not just a single pulse (though we don't 100% know that yet) - but for the time being we can just add a normal measurement (i.e. `qp.meausre(...)` if we want things "to look pretty"

Reply:
Yes, I agree in that is a full block and not just a pulse, but the question for me is to when it is included as a parameter, in execute_anneal() or somewhere else.

Reply:
As we do not know if the S-curve will even happen to be a reality, could we include here for the moment (September demo) a normal measurement pulse? It should be a preloaded pulse, either a square pulse or a flat top, so we should have flexibility on using a pre-calibrated pulse.

Reply:
will be dealt here to avoid making this pr too big
https://linear.app/qilimanjaro/issue/QHC-691/add-default-measurement-pulse-to-annealing-program

Reply:
Agree.